File: DatabaseMicroservice/src/main/java/com/licenta/databasemicroservice/persistence/entity/UserDetails.java
package com.licenta.databasemicroservice.persistence.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.NotEmpty;


@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="user_details")
public class UserDetails {

    @Id
    private Long userId;
    @NotEmpty
    private String firstName;
    @NotEmpty
    private String lastName;

    private String phoneNumber;
    private String address;
    private String profileDescription;
    private String profilePictureUrl;
    private Integer cityId;
}

File: frontend/users/src/app/models/user_details.ts
import { City } from "./city";
import { Country } from "./country";

export interface UserDetails {
  firstName: string;
  lastName: string;
  phoneNumber: string;
  cityId: number;
  address: string;
  profileDescription: string;
  profilePictureUrl: string;
  city?: City;
  country?: Country;
}
